### PR TITLE
feat(usdt-approval): :sparkles: add zero approval flow to limit orders

### LINKED ISSUE #2506: Approve USDT flow for Limit orders
reported in https://github.com/cowprotocol/cowswap/pull/2503#pullrequestreview-1436303082

  # To Reproduce

1. Go to CoW Swap on Goerli, 
2. Use Anxo's USDT https://goerli.etherscan.io/address/0x7b77F953e703E80CD97F6911385c0b1ceabC96Bc
3. Approve or trade to have some left over approvals (for example 5 USDT)
4. Open Limit orders page
5. Enter an amount higher than the approval amount (for example 10 USDT)
6. You should have the two step approval flow
7. This flow should not be shown if it is not necessary (i.e. most other tokens, if you have enough USDT allowance to begin with etc.)

**Actual result:** there is no banner

### PR DESCRIPTION
**Pull request by tukantje**
# Summary

Fixes #2506 

Adds zero approval flow to limit orders.

Screenshot 2023-05-23 at 13 06 38
Screenshot 2023-05-23 at 13 04 22

# To Test

1. Go to CoW Swap on Goerli,
2. Use Anxo's USDT https://goerli.etherscan.io/address/0x7b77F953e703E80CD97F6911385c0b1ceabC96Bc
3. Approve or trade to have some left over approvals (for example 5 USDT)
4. Open Limit orders page
5. Enter an amount higher than the approval amount (for example 10 USDT)
6. You should have the two step approval flow
7. This flow should not be shown if it is not necessary (i.e. most other tokens, if you have enough USDT allowance to begin with etc.)

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @tukantje , great job!
However, 1 nitpick: I see that the warning message apepars on the Limit orders page when I have not yet specified To token. 
image
Could you please fix this?

**Comment by tukantje** [comment]
> Hey @tukantje , great job! However, 1 nitpick: I see that the warning message apepars on the Limit orders page when I have not yet specified To token. image Could you please fix this?

Good point. Technically, you only need to approve for the from token, which is the reason we know it already, but looks a bit odd.

Updated it.

Screenshot 2023-05-23 at 13 37 39

**Review by elena-zh — APPROVED** [review]
LGTM

File: src/common/pure/ZeroApprovalWarning/ZeroApprovalWarning.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ const Link = styled(HashLink)`
 `
 
 interface ZeroApprovalWarningProps {
-  currency: Currency | undefined
+  currency: Currency | undefined | null
```
Comment (alfetopito) [inline src/common/pure/ZeroApprovalWarning/ZeroApprovalWarning.tsx:11]: Use the `Nullish` type https://github.com/cowprotocol/cowswap/blob/b4cdc4ed8a20408d985c6a55776b02905ae71661/src/types/index.ts#L11

**Review by alfetopito — APPROVED** [review]
Very nice!
Only one minor nitpick

File: src/common/pure/ZeroApprovalWarning/ZeroApprovalWarning.tsx (hunk 1 above)
Comment (tukantje) [reply src/common/pure/ZeroApprovalWarning/ZeroApprovalWarning.tsx:11]: Makes sense - updating :+1:

**Review by anxolin — COMMENTED** [review]
👍

**Pull request merged**